### PR TITLE
group component initial changes

### LINKED ISSUE #77: Group Card Component
### Description
We need to create an Event Card Component that matches the design doc found [here](https://www.figma.com/file/mbJusS7ADxfO7s8lkNLc9B/GCColab-HIFI?type=design&node-id=1099-4779).  The event card component should use [mat-card](https://material.angular.io/components/card/overview) for its base with an image that fills 100% of the width and 50% of the top of the card. The bottom half will have centered text that displays the name of the group as well as the group state (closed/open). If the card becomes focused/moused-over the group name should become underlined, and it would be nice if the group picture [zoomed in slightly without changing the image size](https://stackoverflow.com/questions/64889194/how-do-i-zoom-several-images-on-hover-without-changing-its-width-and-height). If a user clicks on the group (at the [mat-card](https://material.angular.io/components/card/overview) level so any part of the card is clickable) we will want to send them to the groups page. Since we haven't setup any subrouting yet we can just send them to the `/groups` route for now. Make sure to use [RouterLink](https://angular.io/api/router/RouterLink) so we don't do a hard refresh. See the header/footer for reference.

Styling will be similar to the news card component (box shadow, border etc.) so feel free to re-use any of the styles found in `src > app > features > news-fed > components > news-card > news-card.component.scss`

### Project Setup
A component called `group-card.component` should be added to the groups feature module found under `src > app > features > groups > components`. Make sure to add the newly created group-card component to the `declarations` and `exports` array of the `groups.module.ts` file so it can be imported into other modules in our project. Once the component is available for export you will want to add the `GroupsModule` to the `imports` array of the `home.module.ts` file found in `src > app > features > home`. You should then be able to use the groups-card component on the home page.

### Mocking data
You should create a new folder called `models` under the groups folder found at `src > app > features > groups`. Similar to the news-feed feature, this will be used to mock a data model that will be used in the` group-card.component` as well as other instances in our app where group is used. The model should be called `Group` and the file named `group.model`, consisting of the following properties:

- id
- name
- displayPicture
- groupStatus

`groupStatus` should be an enum with at least 2 options: `Open` and `Closed`. Add a new file called `group-type.ts` to the same folder (models). See button-type/input-type etc. for reference `src > app > shared > models`.

Since this is mock data and our model isn't set in stone feel free to add any other properties you see fit beside the 4 outlined above.

Once you've setup your models you'll want to instantiate a `group.model` class inside the `home.component.ts` file and fill it with whatever data you want. You can then pass the model to your `group-card.component` as an [Input](https://angular.io/api/core/Input) in the `home.component.html` template and it should populate with the expected data. For reference on Input and the model pattern I've outlined feel free to check out `news-feed-card.component.ts`

### Acceptance Criteria
- Requirements outlined above are met
- Full translated/accessible with [tooltips](https://material.angular.io/components/tooltip/overview) if needed.
- Fills 100% of the width/height assigned at the the top level of the component. Maintains proportions and avoids clipping, ellipses text etc.

### Reference Images

Image

Image

### PR DESCRIPTION
**Pull request by AdiMakkar**
closes #77

### CODE REVIEW

**Comment by AdiMakkar** [comment]
requires changes for: 
1. model
2. [routerLink] 

in group-card.component.html

Additionally to remove comment from the InterceptorGuard before merging

File: src/app/features/groups/components/group-card/group-card.component.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
+import { Group } from '../../models/group'; 
+import { Translations } from 'src/app/core/services/translations.service';
+import { MaterialButtonType } from 'src/app/shared/models/material-button-type';
+import { Router } from '@angular/router';
+import { CoreRoutes } from 'src/app/core/constants/routes.constants';
+import { TooltipDirection } from 'src/app/shared/models/tooltip-direction';
```
Comment (doug0102) [inline src/app/features/groups/components/group-card/group-card.component.ts:7]: I don't think we'll be needing tooltips for this component. You can remove this.

File: src/app/features/groups/components/group-card/group-card.component.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+   templateUrl: './group-card.component.html',
+   styleUrls: ['./group-card.component.scss'],
+   changeDetection: ChangeDetectionStrategy.OnPush
+ })
+ export class GroupCardComponent {
+   @Input() model?: Group;
+   @Input() loading: boolean = false;
+
+   materialButtonType = MaterialButtonType;
+   tooltipDirection = TooltipDirection;
```
Comment (doug0102) [inline src/app/features/groups/components/group-card/group-card.component.ts:20]: I don't think we'll be needing tooltips for this component. You can remove this.

File: src/app/features/groups/components/group-card/group-card.component.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+   routes = CoreRoutes;
+   groupStatus: string = 'Open';
```
Comment (doug0102) [inline src/app/features/groups/components/group-card/group-card.component.ts:22]: This isn't used so you can remove it.

File: src/app/features/groups/components/group-card/group-card.component.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+   selector: 'app-group-card',
+   templateUrl: './group-card.component.html',
+   styleUrls: ['./group-card.component.scss'],
+   changeDetection: ChangeDetectionStrategy.OnPush
+ })
+ export class GroupCardComponent {
+   @Input() model?: Group;
+   @Input() loading: boolean = false;
+
+   materialButtonType = MaterialButtonType;
```
Comment (doug0102) [inline src/app/features/groups/components/group-card/group-card.component.ts:19]: This isn't used so you can remove it.

File: src/app/core/services/group.service.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+            'GCconnex News',
+            'Belly Flop Olympics',
+            'Sleeping Beauties'
+        ];
+        return names[Math.floor(Math.random() * names.length)];
+    }
+
+    private randomGroupStatus(): GroupStatus {
+        const status: GroupStatus[] = [
+            //Open,
```
Comment (doug0102) [inline src/app/core/services/group.service.ts:60]: Replace with

`GroupStatus.Open,
            GroupStatus.Closed`

File: src/app/features/groups/components/group-card/group-card.component.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+   tooltipDirection = TooltipDirection;
+   routes = CoreRoutes;
+   groupStatus: string = 'Open';
+   isFocused: boolean = false;
+
+   constructor(public translations: Translations,
+               private router: Router) 
+   {  }
+
+   onMouseOver() {
```
Comment (doug0102) [inline src/app/features/groups/components/group-card/group-card.component.ts:29]: For the onMouseOver and onMouseOut stuff, is this just to underline text? You can use pure CSS for this using [:hover](https://developer.mozilla.org/en-US/docs/Web/CSS/:hover) on your selector. I would recommend this way as it's less code.

**Review by doug0102 — CHANGES_REQUESTED** [review]

**Review by doug0102 — APPROVED** [review]

**Pull request merged**